tests/counter/counter_test.go

```diff
@@ -0,0 +1,91 @@
+package counter_test
+
+import (
+	"math"
+	"math/big"
+	"testing"
+
+	"github.com/Fantom-foundation/go-opera/tests"
+	"github.com/Fantom-foundation/go-opera/tests/contracts/counter"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+)
+
+func TestCounter(t *testing.T) {
+	net, err := tests.StartIntegrationTestNet(t.TempDir())
+	if err != nil {
+		t.Fatalf("Failed to start the fake network: %v", err)
+	}
+	defer net.Stop()
+
+	t.Run("can increment and read counter from head", func(t *testing.T) {
+
+		// Deploy the counter contract.
+		contract, _, err := tests.DeployContract(net, counter.DeployCounter)
+		if err != nil {
+			t.Fatalf("failed to deploy contract; %v", err)
+		}
+
+		// Increment the counter a few times and check that the value is as expected.
+		for i := 0; i < 10; i++ {
+			counter, err := contract.GetCount(nil)
+			if err != nil {
+				t.Fatalf("failed to get counter value; %v", err)
+			}
+
+			if counter.Cmp(new(big.Int).SetInt64(int64(i))) != 0 {
+				t.Fatalf("unexpected counter value; expected %d, got %v", i, counter)
+			}
+
+			_, err = net.Apply(contract.IncrementCounter)
```

Comment:
would it make sense to check the `receipt.Status` for a success ?

Reply:
I did not want to rewrite the test, just superficial rearrangement. 
but yes. This should be added.